modules/ws/src/main/java/org/mule/module/ws/consumer/WSConsumerConfig.java

```diff
@@ -68,7 +71,7 @@ public void setMuleContext(MuleContext muleContext)
     /**
      * Creates an outbound endpoint for the service address.
      */
-    public MessageProcessor createOutboundMessageProcessor() throws MuleException
+    public MessageProcessor createOutboundMessageProcessor(MessageProcessor messageProcessor) throws MuleException
```

Comment:
keep the original constructor in java and invoke this for backwards compatibility. Remove the changes in the tests below accordingly.